blocks/bricks/sequence_generators.py

```diff
@@ -290,38 +290,81 @@ def readout(self, **kwargs):
         pass
 
 
-@add_metaclass(ABCMeta)
-class Readout(AbstractReadout):
+class Readout(AbstractReadout, Initializable):
     """Readout brick with separated emitting and feedback parts.
 
     Parameters
     ----------
+    source_names : list
+        A list of the source names (outputs) that are needed for the
+        readout part e.g. ``['states']`` or ``['states', 'glimpses']``.
     readout_dim : int
         The dimension of the readout.
     emitter : an instance of :class:`AbstractEmitter`
         The emitter component.
     feedback_brick : an instance of :class:`AbstractFeedback`
         The feedback component.
+    merger : :class:`.Brick`, optional
```

Comment:
It's a bit inconsistent to have `merger` and `post_merge`. They should by identical except for the prefix, IMHO.